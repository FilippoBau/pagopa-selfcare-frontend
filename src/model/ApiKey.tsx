import {ENV} from "../utils/env";

export type ProductKeys = {
    id: string;
    displayName: string;
    primaryKey: string;
    secondaryKey: string;
};

export type AvailableProductKeys = {
    id: string;
    title: string;
    disabled: boolean;
};

export type ConfiguredProductKeys = {
    id: string;
    key: string;
};

export const NODOAUTH = 'Connessione con nodo';
export const GPD = 'GPD - Integrazione Asincrona';
export const GPD_REP = 'GPD - Gestione Flussi di Rendicontazione';
export const BIZ = 'BIZ - Recupero Ricevuta';
export const FDR_ORG = 'FdR - Flussi di Rendicontazione [ORG]';
export const FDR_PSP = 'FdR - Flussi di Rendicontazione [PSP]';

export const API_KEY_PSP_PRODUCTS = (): Array<ConfiguredProductKeys> => {
    let list = [{id: 'NODOAUTH', key: NODOAUTH}];

    if (ENV.FEATURES.FDR) {
        list.push({id: 'FDR_PSP', key: FDR_PSP});
    }
    return list;
};


export const API_KEY_PRODUCTS = (): Array<ConfiguredProductKeys> => {
    let list = [
        {id: 'NODOAUTH', key: NODOAUTH},
        {id: 'GPD_REP', key: GPD_REP},
        {id: 'GPD', key: GPD},
        {id: 'BIZ', key: BIZ},
    ];

    if (ENV.FEATURES.FDR) {
        list.push({id: 'FDR_ORG', key: FDR_ORG});
    }
    return list;
};
