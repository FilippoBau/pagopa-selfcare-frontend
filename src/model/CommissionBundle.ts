export enum FormAction {
  Create = 'create',
  Edit = 'edit',
}
export enum SubscriptionStateType {
  Waiting = 'waiting',
  Accepted = 'accepted'
}

export type PublicBundleCISubscriptionsMethodParams = {
  idBundle: string;
  pspTaxCode: string;
  limit: number;
  page: number;
  status: SubscriptionStateType;
  ciTaxCode?: string;
};

export type PublicBundleCISubscriptionsRequest = {
  "id-bundle": string;
  "psp-tax-code": string;
  limit: number;
  page: number;
  status: SubscriptionStateType;
  ciTaxCode?: string;
};

