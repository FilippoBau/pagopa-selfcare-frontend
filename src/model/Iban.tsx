import { IbanLabel } from '../api/generated/portal/IbanLabel';

export enum IbanFormAction {
  Create = 'create',
  Edit = 'edit',
}

export type IbanOnCreation = {
  iban?: string;
  description?: string;
  validityDate: Date | null;
  dueDate: Date | null;
  creditorInstitutionCode?: string;
  publicationDate?: Date | null;
  ecOwner?: string;
  labels?: Readonly<Array<IbanLabel>>;
};
