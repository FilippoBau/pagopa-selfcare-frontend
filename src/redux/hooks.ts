import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';
import {useHistory} from 'react-router-dom';
import type {RootState, AppDispatch} from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

<<<<<<< HEAD
export const useAppSelectorWithRedirect = ({
  selector,
  routeToRedirect,
  conditionToRedirect,
}: {
  selector: (state: RootState) => any;
  routeToRedirect?: string;
  conditionToRedirect?: (value: any) => boolean;
}) => {
  const selectedReduxState = useAppSelector(selector);
  const history = useHistory();
  const conditionResults = conditionToRedirect ? conditionToRedirect(selectedReduxState) : false;
  if (
    routeToRedirect &&
    (!selectedReduxState ||
      (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0) ||
      conditionResults)
  ) {
    history.push(routeToRedirect);
  }
  return selectedReduxState;
=======
export const useAppSelectorWithRedirect = (
    selector: (state: RootState) => any,
    routeToRedirect?: string
) => {
    const selectedReduxState = useAppSelector(selector);
    const history = useHistory();
    if (
        routeToRedirect &&
        (!selectedReduxState ||
            (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0))
    ) {
        history.push(routeToRedirect);
    }
    return selectedReduxState;
>>>>>>> 3f32cfc3 (Formatting (#542))
};
