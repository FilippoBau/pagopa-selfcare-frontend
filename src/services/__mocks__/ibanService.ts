import { IbansResource } from '../../api/generated/portal/IbansResource';
import { IbanOnCreation } from '../../model/Iban';

export const ibanList: IbansResource = {
  ibanList: [
    {
      ibanValue: 'IT99C0222211111000000000001',
      publicationDate: new Date('2023-06-01T23:59:59.999Z'),
      validityDate: new Date('2023-04-01T13:49:19.897Z'),
    },
    {
      ibanValue: 'IT99C0222211111000000000002',
      publicationDate: new Date('2024-06-01T23:59:59.999Z'),
      validityDate: new Date('2024-04-01T13:49:19.897Z'),
    },
    {
      ibanValue: 'IT99C0222211111000000000003',
      publicationDate: new Date('2023-06-01T23:59:59.999Z'),
      validityDate: new Date('2023-04-01T13:49:19.897Z'),
    },
  ],
};

export const mockedIban = {
  ibanCode: 'IT10T0760116200000014700876',
  ibanDescription: 'Tassa di concorso - servizio tesoreria comunale',
  startDate: new Date('2023-04-01T13:49:19.897Z'),
  endDate: new Date('2033-04-01T13:49:19.897Z'),
  holderFiscalCode: undefined,
};

export const getCreditorInstitutionIbans = (
  _creditorInstitutionCode: string
): Promise<IbansResource> => new Promise((resolve) => resolve(ibanList));

export const getIbanDetail = (_iban: string): Promise<any> =>
  new Promise((resolve) => resolve(ibanList.ibanList[0]));

export const getIbanList = (_istitutionId: string): Promise<any> =>
  new Promise((resolve) => resolve(ibanList));

export const getIban = (iban: string): Promise<any> => new Promise((resolve) => resolve(iban));

export const createIban = (_iban: IbanOnCreation): Promise<any> =>
  new Promise((resolve) => resolve(mockedIban));
