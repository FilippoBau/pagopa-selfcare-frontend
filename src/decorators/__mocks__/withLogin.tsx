import {useEffect} from 'react';
import {User} from '@pagopa/selfcare-common-frontend/model/User';
<<<<<<< HEAD
import {
    userActions,
    userSelectors,
} from '@pagopa/selfcare-common-frontend/redux/slices/userSlice';
=======
import {userActions, userSelectors,} from '@pagopa/selfcare-common-frontend/redux/slices/userSlice';
>>>>>>> 3f32cfc3 (Formatting (#542))
import {useSelector} from 'react-redux';
import {useAppDispatch} from '../../redux/hooks';
import {RootState} from '../../redux/store';

export const mockedUser: User = {
    name: 'NAME',
    surname: 'SURNAME',
    uid: 'UID',
    taxCode: 'AAAAAA00A00A000A',
    email: 'a@a.aa',
};

export const verifyMockExecution = (state: RootState) => {
    expect(state.user.logged).toMatchObject(mockedUser);
};

export default (WrappedComponent: React.ComponentType<any>) => () => {
    const dispatch = useAppDispatch();
    const loggedUser = useSelector(userSelectors.selectLoggedUser);
    useEffect(() => {
        dispatch(userActions.setLoggedUser(mockedUser));
    }, []);
    return loggedUser ? <WrappedComponent/> : <></>;
};
