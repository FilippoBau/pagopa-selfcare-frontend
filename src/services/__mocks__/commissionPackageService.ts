import { Bundles } from '../../api/generated/portal/Bundles';
import { PaymentTypes } from '../../api/generated/portal/PaymentTypes';
import {
  CommissionPackageListResource,
  CommissionPackageOnCreation,
  TaxonomyServicesResource,
  TouchpointsResource,
} from '../../model/CommissionPackage';
import { mockedPaymentTypes, mockedStationsMerged } from '../__mocks__/channelService';

export const mockedCommissionPackagePspList: Bundles = {
  bundles: [
    {
      name: 'Lorem ipsum',
      validity_date_from: new Date(),
      validity_date_to: new Date(),
      touchpoint: 'Checkout',
      payment_type: mockedPaymentTypes?.payment_types?.[0]?.payment_type,
      min_payment_amount: 0,
      max_payment_amount: 150,
    },
  ],
  page_info: {
    items_found: 1,
    limit: 10,
    page: 0,
    total_pages: 1,
  },
};

export const mockedTouchpoints: TouchpointsResource = {
  touchpointList: [{ touchpoint: 'Tutti' }, { touchpoint: 'App Io' }, { touchpoint: 'Checkout' }],
};

export const mockedTaxonomy: TaxonomyServicesResource = {
  taxonomyServiceList: [{ taxonomyService: '100 - Rendite catastali (ICI, IMU, TUC, ecc.)' }],
};

export const mockedCommissionPackagePspDetail: CommissionPackageOnCreation = {
  abi: '12345',
  description: 'Pacchetti commissione',
  digitalStamp: true,
  digitalStampRestriction: false,
  idBrokerPsp: '12345UK6789',
  idCdi: '12345UK6789',
  idChannel: '97735020584_01',
  maxPaymentAmount: 1500,
  minPaymentAmount: 150,
  name: 'Pacchetto 1',
  paymentAmount: 10,
  paymentType: mockedPaymentTypes!.payment_types![0],
  touchpoint: mockedTouchpoints,
  transferCategoryList: ['100 - Rendite catastali (ICI, IMU, TUC, ecc.) '],
  type: 'GLOBAL',
  validityDateFrom: new Date(),
  validityDateTo: new Date(),
};

export const mockedChannelsIdList: Array<string> = mockedStationsMerged!.channels!.map((e) =>
  typeof e.channel_code !== 'undefined' ? e.channel_code : ''
);

export const getChannelsId = (_page: number, _brokerCode: string): Promise<Array<string>> =>
  new Promise((resolve) => resolve(mockedChannelsIdList));

export const getCommissionPackagePsp = (
  _brokerCode: string
): Promise<Bundles> =>
  new Promise((resolve) => resolve(mockedCommissionPackagePspList));

export const getPaymentTypes = (): Promise<PaymentTypes> =>
  new Promise((resolve) => resolve(mockedPaymentTypes));

export const getCommissionPackageDetails = (_name: string): Promise<CommissionPackageOnCreation> =>
  new Promise((resolve) => resolve(mockedCommissionPackagePspDetail));

export const createCommissionPackage = (
  _body: CommissionPackageOnCreation
): Promise<CommissionPackageOnCreation> =>
  new Promise((resolve) => resolve(mockedCommissionPackagePspDetail));

export const updateCommissionPackage = (
  _name: string,
  _commissionPackage: CommissionPackageOnCreation
): Promise<CommissionPackageOnCreation> =>
  new Promise((resolve) => resolve(mockedCommissionPackagePspDetail));

export const getTouchpoint = (): Promise<TouchpointsResource> =>
  new Promise((resolve) => resolve(mockedTouchpoints));

export const getTaxonomyService = (): Promise<TaxonomyServicesResource> =>
  new Promise((resolve) => resolve(mockedTaxonomy));
