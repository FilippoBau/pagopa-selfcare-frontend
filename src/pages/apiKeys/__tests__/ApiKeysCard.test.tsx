import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { theme } from '@pagopa/mui-italia';
import { ThemeProvider } from '@mui/system';
import routes from '../../../routes';
import ApiKeysCard from '../ApiKeysCard';
import { getInstitutionApiKeys } from '../../../services/__mocks__/apiKeyService';
import { ProductKeys } from '../../../model/ApiKey';

//SNAPSHOT TESTING
it('renders correctly', () => {
  getInstitutionApiKeys('').then((data) => {
    const tree = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[routes.APIKEYS]}>
          <Route path={routes.APIKEYS}>
            <ThemeProvider theme={theme}>
              <ApiKeysCard apiKey={data[0]} />
            </ThemeProvider>{' '}
          </Route>
        </MemoryRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
