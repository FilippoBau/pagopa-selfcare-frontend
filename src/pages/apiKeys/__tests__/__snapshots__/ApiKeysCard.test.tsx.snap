// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-19b52s8-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-3grx0y-MuiTypography-root"
          >
            Connessione con nodo
          </h6>
        </div>
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-3c3y12-MuiTypography-root"
          >
            apiKeysPage.apiPresent.primaryApiKey
          </p>
          <div
            class="css-9jbana-MuiStack-root"
          >
            <div
              class="MuiBox-root css-yq76c3"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-11o4oo8-MuiTypography-root"
              >
                XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
              </p>
            </div>
            <button
              aria-pressed="true"
              class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-18a3qz2-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="check"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1l2o29s-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1ft2xoq-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              apiKeysPage.apiPresent.useKeyBtn
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-19ja3ml-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              apiKeysPage.apiPresent.regeneratesBtn
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-3c3y12-MuiTypography-root"
          >
            apiKeysPage.apiPresent.secondaryApiKey
          </p>
          <div
            class="css-9jbana-MuiStack-root"
          >
            <div
              class="MuiBox-root css-yq76c3"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-11o4oo8-MuiTypography-root"
              >
                XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
              </p>
            </div>
            <button
              aria-pressed="true"
              class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-18a3qz2-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="check"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1l2o29s-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1ft2xoq-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              apiKeysPage.apiPresent.useKeyBtn
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-19ja3ml-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              apiKeysPage.apiPresent.regeneratesBtn
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-19b52s8-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-1qm1lh"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-3grx0y-MuiTypography-root"
        >
          Connessione con nodo
        </h6>
      </div>
      <div
        class="MuiBox-root css-1qm1lh"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-3c3y12-MuiTypography-root"
        >
          apiKeysPage.apiPresent.primaryApiKey
        </p>
        <div
          class="css-9jbana-MuiStack-root"
        >
          <div
            class="MuiBox-root css-yq76c3"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-11o4oo8-MuiTypography-root"
            >
              XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
            </p>
          </div>
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-18a3qz2-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="check"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1l2o29s-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1ft2xoq-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            apiKeysPage.apiPresent.useKeyBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-19ja3ml-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            apiKeysPage.apiPresent.regeneratesBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-3c3y12-MuiTypography-root"
        >
          apiKeysPage.apiPresent.secondaryApiKey
        </p>
        <div
          class="css-9jbana-MuiStack-root"
        >
          <div
            class="MuiBox-root css-yq76c3"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-11o4oo8-MuiTypography-root"
            >
              XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
            </p>
          </div>
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard css-18a3qz2-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="check"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1l2o29s-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1ft2xoq-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            apiKeysPage.apiPresent.useKeyBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-19ja3ml-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            apiKeysPage.apiPresent.regeneratesBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
